src/index.js

```diff
@@ -122,21 +122,18 @@ class DeltaUpdater extends EventEmitter {
     this.deltaHolderPath = path.join(this.deltaUpdaterRootPath, './deltas');
 
     if (app.isPackaged && process.platform === 'darwin') {
-      this.macUpdaterPath = path.join(this.deltaUpdaterRootPath, 'mac-updater');
-      this.hpatchzPath = path.join(this.deltaUpdaterRootPath, 'hpatchz');
-      // https://developer.apple.com/forums/thread/130313?answerId=410704022#410704022
-      fs.moveSync(macUpdaterPath, this.macUpdaterPath, {
-        overwrite: true,
-      });
-      fs.moveSync(hpatchzPath, this.hpatchzPath, {
-        overwrite: true,
-      });
+      this.macUpdaterPath = path.join(this.deltaUpdaterRootPath, './mac-updater');
+      this.hpatchzPath = path.join(this.deltaUpdaterRootPath, './hpatchz');
+      await downloadFile(macUpdaterURL, this.macUpdaterPath);
+      await downloadFile(hpatchzURL, this.hpatchzPath);
+      fs.chmodSync(this.macUpdaterPath, '755');
```

Comment:
I think you could use the promisified version of chmod, that way it won't block the app during these checks. https://nodejs.org/api/fs.html#fspromiseschmodpath-mode